### PR TITLE
added Organization.globalConstraints

### PR DESCRIPTION
**Pull request by dkent600**
This addresses #28 - add getGlobalConstraints function to Organization
and #30 - Remove migrateContracts.initialize script

It is required to be able to proceed with a PR for https://github.com/daostack/Alchemy-Aurelia/issues/11

### CODE REVIEW

File: lib/organization.js [hunk 1]
```diff
@@ -145,24 +145,22 @@ export class Organization {
...
-      });
+      return schemes.filter(s => s.name === name);
     } else {
       return schemes;
     }
   }
 
   /**
-   * returns schemes currently in this Organization as Array<OrganizationSchemeInfo>
+   * returns schemes currently in this DAO as Array<OrganizationSchemeInfo>
```
Comment (tibetsprague) [inline lib/organization.js]: why this change to DAO? seems confusing to use both, we should pick one or the other

File: lib/organization.js [hunk 2]
```diff
@@ -273,4 +271,97 @@ export class Organization {
...
+    const constraints = await this._getConstraints();
+    if (name) {
+      return constraints.filter(s => s.name === name);
+    } else {
+      return constraints;
+    }
+  }
+
+  /**
+   * returns global constraints currently in this Organization, as OrganizationGlobalConstraintInfo
```
Comment (tibetsprague) [inline lib/organization.js]: Organization or DAO?

File: lib/organization.js [hunk 3]
```diff
@@ -273,4 +271,97 @@ export class Organization {
...
+   */
+  async _getConstraints() {
+    // private method returns all registered schemes.
```
Comment (tibetsprague) [inline lib/organization.js]: constraints not schemes, also why do we need this comment alongside the block comment above the function?

File: lib/organization.js [hunk 4]
```diff
@@ -273,4 +271,97 @@ export class Organization {
...
+      const paramsHash = eventsArray[i].args._params;
+
+      const info = {
+        address: address,
+        paramsHash: paramsHash,
+        // will be undefined if not a known scheme
+        name: arcTypesMap.get(address)
+      };
+
+      constraintsMap.set(address, info);
```
Comment (tibetsprague) [inline lib/organization.js]: why not just check if its registered here instead of adding all of them to the Map and then iterating through the whole thing again to see which ones are registered?

File: test/organization.js [hunk 5]
```diff
@@ -76,4 +77,67 @@ describe("Organization", () => {
 
     assert.equal((await organization.schemes()).length, 4);
   });
+
+  it("has a working globalConstraints() function to access its schemes", async () => {
```
Comment (tibetsprague) [inline test/organization.js]: confusing to use the word schemes here

File: lib/organization.js (hunk 3 above)
Comment (dkent600) [reply lib/organization.js]: yeah, there are a lot of old comments in this code from long before my time that I tend to just block out of my mind...

File: lib/organization.js (hunk 4 above)
Comment (dkent600) [reply lib/organization.js]: the map eliminates dups

File: lib/organization.js (hunk 1 above)
Comment (dkent600) [reply lib/organization.js]: agreed, done

File: lib/organization.js (hunk 2 above)
Comment (dkent600) [reply lib/organization.js]: done

File: lib/organization.js (hunk 3 above)
Comment (dkent600) [reply lib/organization.js]: deleted

File: lib/organization.js (hunk 4 above)
Comment (dkent600) [reply lib/organization.js]: commented

File: test/organization.js (hunk 5 above)
Comment (dkent600) [reply test/organization.js]: done

File: lib/organization.js (hunk 4 above)
Comment (tibetsprague) [reply lib/organization.js]: so you think its more likely that there will be dupes to eliminate first rather than unregistered constraints to eliminate first? either way its not a big deal
